libcaf_net/caf/net/socket.hpp

```diff
@@ -24,20 +24,55 @@
 
 #include "caf/config.hpp"
 #include "caf/fwd.hpp"
-#include "caf/net/abstract_socket.hpp"
 #include "caf/net/socket_id.hpp"
 
 namespace caf {
 namespace net {
 
 /// An internal endpoint for sending or receiving data. Can be either a
 /// ::network_socket, ::pipe_socket, ::stream_socket, or ::datagram_socket.
-struct socket : abstract_socket<socket> {
-  using super = abstract_socket<socket>;
+struct socket {
+  socket_id id;
 
-  using super::super;
+  constexpr socket() : id(invalid_socket_id) {
+    // nop
+  }
+
+  constexpr explicit socket(socket_id id) : id(id) {
+    // nop
+  }
+
+  constexpr socket(const socket& other) : id(other.id) {
+    // nop
+  }
+
+  socket& operator=(const socket& other) {
+    id = other.id;
+    return *this;
+  }
 };
 
+/// @relates socket
+template <class Inspector>
+typename Inspector::result_type inspect(Inspector& f, socket& x) {
+  return f(x.id);
+}
+
+/// @relates socket
+constexpr bool operator==(socket x, socket y) {
```

Comment:
Why not use `const socket&` for these and the following parameters?

Reply:
There is no reason to pass small objects by reference. Passing by value is both less noisy and can be more efficient. However, defining the comparison operators by hand is nonsense, we have `detail::comparable` for that.

Reply:
Even better. Thanks for the explanation! 👍